cr-02-08-2022

Ami777 opened the pull request.

RadekJ87 committed:
Merge pull request #12 from Fremen1990/feature/db-to-test

new entities and tests

Fremen1990 committed:
Changes in cors due to communication with Front-end, added and tested JWT guards, added getCuurentProfile in user service to download user profile on front-end login.

Fremen1990 committed:
Merge pull request #13 from Fremen1990/feature/Jwt-guards-cors-fe

Changes in cors due to communication with Front-end, added and tested…

Fremen1990 committed:
Endpoint generating Random students for development purposes, enum changes

RadekJ87 committed:
refactor enums and fix updateStudentDetails

RadekJ87 committed:
Merge branch 'develop' into feature/random-students100-create-admin

RadekJ87 committed:
Merge pull request #14 from Fremen1990/feature/random-students100-create-admin

Endpoint generating Random students for development purposes, enum ch…

RadekJ87 committed:
testing hr module with new functionalities for interwiev - added getCandidates, getOneCandidate

To do add to interview, remove, hire

Fremen1990 committed:
HOTFIX - Error handling regarding FE errors, changes in endpoints return data

Fremen1990 committed:
Fresh POSTMAN collection with all scenarios

Ami777 commented on src/app.service.ts:
```diff
@@ -1,8 +1,45 @@
+      };
+   }
+
+   async createAdminUser(newAdmin: AdminDto) {
```
Nie wystawiajcie tego na zewnątrz proszę. Niebezpieczne

Ami777 commented on src/enums/apprenticeship.enum.ts:
```diff
@@ -1,15 +1,15 @@
 export const validateApprenticeship = (agreement: string): string => {
-   if (agreement.toUpperCase() === Apprenticeship.yes) {
-      return 'YES';
+   if (agreement.toLowerCase() === Apprenticeship.YES) {
```
Zupełnie nie rozumiem sensu tej funkcji 🤔

Ami777 commented on src/enums/contract-type.enum.ts:
```diff
@@ -1,22 +1,22 @@
 export const validateContractType = (contactType: string): string => {
-   if (contactType.toUpperCase() === ContactType.uop) {
-      return 'UOP';
+   if (contactType.toLowerCase() === ContactType.UOP) {
```
Po prostu zróbcie sobie stringowe enumy. Będzie można zrobić `ContactType[contactType]`. Chyba nawet tak macie. Więc nie ma sensu robić tego tak na sztywno.

RadekJ87 replied:
Dzięki Kuba, wczoraj zdałem sobie sprawę, że nie za bardzo to coś daje, ale z drugiej strony zapomniałem to usunąć 😉

Ami777 commented on src/interfaces/user.ts:
```diff
@@ -19,3 +21,18 @@ export interface createOneUserResponse {
+   role: string;
+   currentSessionToken: string | null;
+   active: boolean;
+   created_at: Date;
```
Dlaczego `jedneZmienneTak`, a `inne_zmienne_tak`?

RadekJ87 replied:
Ja pisałem , Tomek pisał. Nikt nie zauważył🙂

Ami777 reviewed (commented):
Jest ok, wrzuciłem kilka ważnych podpowiedzi

Fremen1990 committed:
HotFix - change in getCurrentUserProfile to the same return for all roles

Fremen1990 committed:
ReadMe update, newman tests and postman collection update

Fremen1990 committed:
PostMan Tests and bufixes regarding to found errors with postman

Fremen1990 committed:
PostMan Tests FINAL

Fremen1990 committed:
Swagger documentation description and compodoc installation

RadekJ87 committed:
Final database with relations - tested.
Added method - addOneCandidateToList

RadekJ87 committed:
Merge pull request #17 from Fremen1990/feature/55-Swagger-Compodoc

Swagger documentation description and compodoc installation

Fremen1990 committed:
Merge branch 'develop' into feature/interview

Fremen1990 committed:
Merge pull request #15 from Fremen1990/feature/interview

testing hr module with new functionalities for interview

Fremen1990 committed:
Formatting changes after conflicts resolution

Fremen1990 committed:
Export types to FE in index.ts

Fremen1990 committed:
Corrections for client DEMO

Fremen1990 committed:
Postaman Update

RadekJ87 committed:
Refactor of getCandidateList(), getOneCandidate(), getOneHr, addOneCandidateToList(). Add method showMyInteviews()

RadekJ87 committed:
final HR section

RadekJ87 committed:
up metohods in controller

RadekJ87 committed:
Merge pull request #18 from Fremen1990/feature/refactor-interview

Refactor

Fremen1990 committed:
Change @IsString@ to IsArray on bonusProjectUrls

Fremen1990 committed:
Send change password and reset password not protected jwt

RadekJ87 committed:
pwd keys moved to env
fixed funtionality of send reset token and change password
next-cli fix - problem with dir mail/template
mailservice - added new email funcitonality sendResetToken
Student Entity - changed length of phone number to 20

RadekJ87 committed:
added validation for max reserved students

Fremen1990 committed:
Working mail activation and redirection to login page

Fremen1990 committed:
POSTMAN Collection V3

Fremen1990 committed:
Emails layout with logo

Fremen1990 committed:
Student First Login entity collumn and first login backend functionality

RadekJ87 committed:
refactor showMyInterviews()

RadekJ87 committed:
Merge remote-tracking branch 'origin/develop' into develop

RadekJ87 committed:
nest-cli quick fix for mail assets

RadekJ87 committed:
Cron service:
 -- every day at 01:30 cron runs method that is checking if already 10 days passed from adding to interview and removes matching records from interview list

Fremen1990 committed:
Export types to front-end

Fremen1990 committed:
Merge remote-tracking branch 'origin/develop' into develop

RadekJ87 committed:
Merge pull request #19 from Fremen1990/feature/cron

Cron service:

RadekJ87 committed:
- getCandidatesList() is showing students with status interview too,
- getOneCandidate() is showing student with status interview too,
- addOneCandidateToList() is preventin from adding a student twice,                                                               if student is added to interview for the first time changes the status on 'interview'.
- remove() - added funtionality if student is no longer in other interviews the student status is changed back to 'available'.
- hire() - added funtionality if student is hire then is removed from all other open interview.

Created new methods - cleanUpAfterHiring(), getStudentInterviewsCount(), doubleInterviewCheck()

Fremen1990 committed:
Guard JWT removed from "logout"

Fremen1990 committed:
Merge remote-tracking branch 'origin/develop' into develop

RadekJ87 committed:
another refactor of getCandidateList() - fix showing students added to hr's interview

Fremen1990 committed:
RolesGuards done

Fremen1990 committed:
RolesGuards disabled for student CV

RadekJ87 committed:
Merge branch 'develop' into feature/RolesGuards-complete

RadekJ87 committed:
Merge pull request #21 from Fremen1990/feature/RolesGuards-complete

RolesGuards done

RadekJ87 committed:
Merge pull request #20 from Fremen1990/feature/refactor-hr-service-new-requirements

Refactor

RadekJ87 committed:
cron refactor after changes in hr module

Fremen1990 committed:
ReadMe Update

Fremen1990 committed:
Endpoint to toggle student availability hires/available

RadekJ87 committed:
filters for HR

RadekJ87 committed:
Merge pull request #22 from Fremen1990/feature/filter

filters for HR

Fremen1990 committed:
Student Status endpoint response update

Fremen1990 committed:
Merge remote-tracking branch 'origin/develop' into develop

RadekJ87 committed:
filter metods changed from GET to POST

MS151994 committed:
fix destructuring object from FE
fix min max property of salary default value

MS151994 committed:
change value from string to Number

MS151994 committed:
change value from string to Number

MS151994 committed:
change value from string to Number

RadekJ87 committed:
Merge pull request #24 from Fremen1990/chore/fix_hr_filtering

Chore/fix hr filtering

RadekJ87 committed:
refactor studentUpdate - if status interview || available changed to employed and removes from openinterviews, from employed changes to available

RadekJ87 committed:
Created interface: GetOneHrProfileResponse, GetOneHrInterviewsResponse
Created DTO: HrFilterResultsDto

DTO and interfaces implemented to hr service

RadekJ87 committed:
Merge pull request #25 from Fremen1990/feature/dto-and-interfaces-fix

Feature/dto and interfaces fix
